cunumeric/module.py

```diff
@@ -1392,6 +1414,214 @@ def squeeze(a: ndarray, axis: Optional[NdShapeLike] = None) -> ndarray:
+    return arrays
+
+
+def broadcast_arrays(
+    *args: Sequence[Any], subok: bool = False
+) -> list[ndarray]:
+    """
+
+    Broadcast any number of arrays against each other.
+
+    Parameters
+    ----------
+    `*args` : array_likes
+        The arrays to broadcast.
+
+    subok : bool, optional
+        If True, then sub-classes will be passed-through,
+        otherwise the returned arrays will be forced to
+        be a base-class array (default).
+
+    Returns
+    -------
+    broadcasted : list of arrays
+        These arrays are views on the original arrays.
+        They are typically not contiguous.
+        Furthermore, more than one element of a broadcasted array
+        may refer to a single memory location.
+        If you need to write to the arrays, make copies first.
+        While you can set the writable flag True,
+        writing to a single output value may end up changing
+        more than one location in the output array.
+
+    Availability
+    --------
+    Multiple GPUs, Multiple CPUs
+
+    """
+    arrs = [convert_to_cunumeric_ndarray(arr) for arr in args]
+    return _broadcast_arrays(arrs, subok=subok)
+
+
+class broadcast:
+    def __init__(self, *arrays: Sequence[Any]) -> None:
+        """
+
+        Produce an object that broadcasts input parameters
+        against one another. It has shape and nd properties,
+        and may be used as an iterator.
+
+        Parameters
+        ----------
+        `*arrays` : array_likes
+            The arrays to broadcast.
+
+        subok : bool, optional
+            If True, then sub-classes will be passed-through,
+            otherwise the returned arrays will be forced to
+            be a base-class array (default).
+
+        """
+        arrs = [convert_to_cunumeric_ndarray(arr) for arr in arrays]
+        broadcasted = _broadcast_arrays(arrs)
+        self._iters = tuple(arr.flat for arr in broadcasted)
+        self._index = 0
+        self._shape = broadcasted[0].shape
+        self._size = np.prod(self.shape, dtype=int)
+
+    def __iter__(self) -> broadcast:
+        self._index = 0
+        return self
+
+    def __next__(self) -> Any:
+        if self._index < self.size:
+            result = tuple(each[self._index] for each in self._iters)
+            self._index += 1
+            return result
+
+    def reset(self) -> None:
+        self._index = 0
+
+    @property
+    def index(self) -> int:
+        return self._index
+
+    @property
+    def iters(self) -> Tuple[Iterable[Any], ...]:
+        return self._iters
+
+    @property
+    def numiter(self) -> int:
+        return len(self._iters)
+
+    @property
+    def nd(self) -> int:
+        return self.ndim
+
+    @property
+    def ndim(self) -> int:
+        return len(self.shape)
+
+    @property
+    def shape(self) -> NdShape:
+        return self._shape
+
+    @property
+    def size(self) -> int:
+        return self._size
+
+
```

Comment:
FWIW classes like this are things I would personally like to see discrete unit tests for, but I guess that needs to be a larger team discussion.

Reply:
Then, let's keep it as is and let another PR to add the tests to all correpsonding tests.